csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -1520,20 +1520,47 @@ private void TestModelInputUINT64()
             }
         }
 
-        [Fact(Skip = "FLOAT16 not available in C#")]
+        [Fact]
         private void TestModelInputFLOAT16()
         {
             // model takes 1x5 input of fixed type, echoes back
-            string modelPath = Path.Combine(Directory.GetCurrentDirectory(), "test_types_FLOAT16.pb");
+            string modelPath = Path.Combine(Directory.GetCurrentDirectory(), "test_types_FLOAT16.onnx");
             using (var session = new InferenceSession(modelPath))
             {
                 var container = new List<NamedOnnxValue>();
-                var tensorIn = new DenseTensor<float>(new float[] { 1.0f, 2.0f, -3.0f, float.MinValue, float.MaxValue }, new int[] { 1, 5 });
+                var tensorIn = new DenseTensor<Float16>(
+                    new Float16[] { 15360, 16384, 16896, 17408, 17664 }, new int[] { 1, 5 });
```

Comment:
So the users still need to use third party libraries to obtain the corresponding ushort values from their half precision floats ? I thought that the added structs would have capability to arrive at the ushort values (half precision float) from a given single precision / double precision float

Reply:
None of our APIs provide any conversions.

---
In reply to: [521762473](https://github.com/microsoft/onnxruntime/pull/5775#discussion_r521762473) [](ancestors = 521762473)